dart/lib/src/sentry_client.dart

```diff
@@ -11,19 +11,24 @@ import 'version.dart';
 /// Logs crash reports and events to the Sentry.io service.
 class SentryClient {
   /// Instantiates a client using [SentryOptions]
-  SentryClient(SentryOptions options) : _options = options {
-    _random = _options.sampleRate == null ? null : Random();
-    if (_options.transport is NoOpTransport) {
-      _options.transport = HttpTransport(options: _options);
+  factory SentryClient(SentryOptions options) {
+    if (options.transport is NoOpTransport) {
+      options.transport = HttpTransport(options);
     }
+    return SentryClient._(options);
   }
 
   final SentryOptions _options;
 
-  Random _random;
+  final Random _random;
 
   static final _sentryId = Future.value(SentryId.empty());
 
+  /// Instantiates a client using [SentryOptions]
```

Comment:
Why did you add a private ctor ?

Reply:
because we have a public factory, so we can separate a bit more the things